bin/crystal

```diff
@@ -7,13 +7,18 @@
 
 set -euo pipefail # exit on any error, don't allow undefined variables, pipes don't swallow errors
 
-# Set up SHARDS_INSTALL_PATH.
-# shellcheck source=bin/shards
-. "$HOME/code/dotfiles/bin/shards"
+# Export a CRYSTAL_PATH for the current directory (if no CRYSTAL_PATH is yet
+# defined).
+if [ -z "${CRYSTAL_PATH+x}" ] ; then
```

Comment:
Respected an already-provided `CRYSTAL_PATH` allows us to specify a `CRYSTAL_PATH` in `bin/run-crystal-program` that loads from the `dotfiles` shards (rather than whichever directory we were in when the compilation was triggered).